converter/converter.py

```diff
@@ -0,0 +1,202 @@
+        detect_strings = [str(geo)
+                          for geo in self.geometries] + [self.detect_template]
+        return "\n".join(detect_strings)
+
+
+@ dataclass
+class GeoConfig:
+    """Class mapping of the geo.dat config file."""
+
+    geo_template: str = """
+*---><---><--------><------------------------------------------------>
+    0    0           protons, H2O 30 cm cylinder, r=10, 1 zone
+*---><---><--------><--------><--------><--------><--------><-------->
+  RCC    1       0.0       0.0       0.0       0.0       0.0      30.0
+                10.0
+  RCC    2       0.0       0.0      -5.0       0.0       0.0      35.0
+                15.0
+  RCC    3       0.0       0.0     -10.0       0.0       0.0      40.0
+                20.0
+  END
+  001          +1
+  002          +2     -1
+  003          +3     -2
+  END
+* material codes: 1 - liquid water (ICRU material no 276), 1000 - vacuum, 0 - black body
+    1    2    3
+    1 1000    0
+"""
+
+    def __str__(self) -> str:
+        return self.geo_template
+
+
+class Parser(ABC):
+    """Abstract parser, the template for implementing other parsers."""
+
+    @ abstractmethod
+    def parse_configs(json: dict):
+        """Convert the json dict to the 4 config dataclasses."""
+        pass
+
+    @ abstractmethod
+    def save_configs(target_dir_path: str):
+        """
+        Save the configs as text files in the target_dir.
+        The files are: beam.dat, mat.dat, detect.dat and geo.dat.
+        """
+        pass
+
+
+class DummmyParser(Parser):
+    """A simple placeholder parser that ignores the json input and prints example (default) configs."""
+
+    def __init__(self) -> None:
+        self.beam_config = BeamConfig()
+        self.mat_config = MatConfig()
+        self.detect_config = DetectConfig()
+        self.geo_config = GeoConfig()
+
+    def parse_configs(self, json: dict):
+        """Basicaly do nothing since we work on defaults in this parser."""
+        pass
+
+    def save_configs(self, target_dir: str):
+        """
+        Save the configs as text files in the target_dir.
+        The files are: beam.dat, mat.dat, detect.dat and geo.dat.
+        """
+        with open(path.join(target_dir, 'beam.dat'), 'w') as beam_f:
+            beam_f.write(self.beam_config)
+
+        with open(path.join(target_dir, 'mat.dat'), 'w') as mat_f:
+            mat_f.write(self.mat_config)
+
+        with open(path.join(target_dir, 'detect.dat'), 'w') as detect_f:
+            detect_f.write(self.detect_config)
+
+        with open(path.join(target_dir, 'geo.dat'), 'w') as geo_f:
+            beam_f.write(self.geo_config)
+
+
+class Runner:
+    """Converts input data dict to files that will be saved in output_dir using the specified parser."""
+
+    def __init__(self, parser: Parser, input_data: dict, output_dir: str) -> None:
+        self.parser = parser
+        self.input_data = input_data
+        self.output_dir = output_dir
+
+    def run_parser(self):
```

Comment:
add type hinting for return type

Reply:
Resolved in b951def677fac830fd2757dc6fb68fb4a8c5e5e5